pi-top/Wi-Fi-Access-Point-and-Station-Mode
# Update default AP SSID & passphrase

| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/OS-1182) |


#### Main changes

Updated default access point credentials:
- Hostname is a mix of the hostname and the last 4 characters of the device serial number. eg: `pi-top-1987`
- Passphrase is a random 8 digit number which is set the first time the access point starts.

#### Screenshots (feature, test output, profiling, dev tools etc)

N/A

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)

N/A

#### Manual testing tips

- Download deb files and scp them into your pi-top
- ssh into your pi-top and install the deb files
- Check the current `hostapd` configuration.
```
pi@pi-top:~ $ cat /etc/hostapd/hostapd.conf
auth_algs=1
channel=6
country_code=00
ctrl_interface=/var/run/hostapd
ctrl_interface_group=0
hw_mode=g
ignore_broadcast_ssid=0
macaddr_acl=0
rsn_pairwise=CCMP
wpa=2
wpa_pairwise=TKIP
interface=wlan_ap0
ssid=pi-top
wpa_passphrase=pi-top1234
```
- Re-configure the AP. Run `sudo wifi-ap-sta configure`.
- Check the `hostapd` configuration again. You should be able to see the new `SSID` & `wpa_passphrase`.
```pi@pi-top:~ $ cat /etc/hostapd/hostapd.conf
auth_algs=1
channel=6
country_code=00
ctrl_interface=/var/run/hostapd
ctrl_interface_group=0
hw_mode=g
ignore_broadcast_ssid=0
macaddr_acl=0
rsn_pairwise=CCMP
wpa=2
wpa_pairwise=TKIP
interface=wlan_ap0
ssid=pi-top-E89E
wpa_passphrase=80691386
```
- This command only updated the configuration files but the AP that's running doesn't know about this. Run `sudo wifi-ap-sta start` to restart AP mode; note that your ssh might hang.
- Try to connect to the AP with the new credentials.
- Verify that the file `/etc/default/wifi-ap-sta` contains the password. This file is used to persist the password on reboot.
```
pi@pi-top:~ $ cat /etc/default/wifi-ap-sta
AP_PASSWORD=80691386
```
- Also, you can try rebooting your device and verifying that the `wpa_passphrase`/`AP_PASSWORD` hasn't changed in `hostapd.conf` & `/etc/default/wifi-ap-sta`.

#### Tag anyone who definitely needs to review or help

N/A
